src/NSwag.SwaggerGeneration.AspNetCore/AspNetCoreToSwaggerGenerator.cs

```diff
@@ -69,6 +69,20 @@ public async Task<SwaggerDocument> GenerateAsync(ApiDescriptionGroupCollection a
                 .GroupBy(item => item.Item2.ControllerTypeInfo.AsType())
                 .ToArray();
 
+            document.Consumes = apiGroups
+                .SelectMany(s => s.SupportedRequestFormats)
+                .Select(s => s.MediaType)
+                .Where(m => apiGroups.All(a => !a.SupportedRequestFormats.Any() || a.SupportedRequestFormats.Contains(m)))
+                .Distinct()
+                .ToList();
+            
+            document.Produces = apiGroups
+                .SelectMany(c => c.SupportedResponseTypes)
+                .SelectMany(s => s.ApiResponseFormats.Select(f => f.MediaType))
+                .Where(c => apiGroups.All(o => !o.SupportedResponseTypes.Any() || o.SupportedResponseTypes.Contains(c)))
```

Comment:
Same here with !any. Is this actually solving the problem?

Reply:
Fixed